File: aqa_js/automation_playwright/pom/addcontactpage.js
const {Page, expect} = require('@playwright/test');
const BasePage = require('./basepage');

class ContactListPage extends BasePage {
    /**
     * @param {Page} page - Playwright Page object
     */
    constructor(page) {
        super(page);
        this.PageUrl = `${this.config.BASE_URL}/addContact`; // Replace with the actual URL of the signup page
        this.ids = {
            logoutButtonId: "#logout",
            firstNameFieldId: "#firstName",
            lastNameFieldId: "#lastName",
            emailFieldId: "#email",
            birthdateFieldId: "#birthdate",
            phoneFieldId: "#phone",
            street1AddressFieldId: "#street1",
            street2AddressFieldId: "#street2",
            cityFieldId: "#city",
            stateFieldId: "#stateProvince",
            postalCodeFieldId: "#postalCode",
            countryFieldId: "#country",
            submitButtonId: '#submit',
            cancelButtonId: '#cancel',
            errorMessageId: '#error'
        };
        this.contactSamples
    }

    async navigate() {
        await super.go_to_url(this.PageUrl);
    }
    async enter_first_name(firstNameText) {
        await this.page.fill(this.ids.firstNameFieldId, firstNameText);
    }
    async enter_last_name(lastNameText) {
        await this.page.fill(this.ids.lastNameFieldId, lastNameText);
    }   
    async enter_email(emailText) {
        await this.page.fill(this.ids.emailFieldId, emailText);
    }
    async enter_birthdate(birthdateText) {
        await this.page.fill(this.ids.birthdateFieldId, birthdateText);
    }   
    async enter_phone(phoneText) {
        await this.page.fill(this.ids.phoneFieldId, phoneText);
    }
    async enter_street1(street1Text) {
        await this.page.fill(this.ids.street1AddressFieldId, street1Text);
    }
    async enter_street2(street2Text) {
        await this.page.fill(this.ids.street2AddressFieldId, street2Text);
    }
    async enter_city(cityText) {
        await this.page.fill(this.ids.cityFieldId, cityText);
    }
    async enter_state(stateText) {
        await this.page.fill(this.ids.stateFieldId, stateText);
    }
    async enter_postal_code(postalCodeText) {
        await this.page.fill(this.ids.postalCodeFieldId, postalCodeText);
    }
    async enter_country(countryText) {
        await this.page.fill(this.ids.countryFieldId, countryText);
    }
    async click_add_contact() {
        await this.page.click(this.ids.addContactButtonId);
    }
    async click_submit() {
        await this.page.click(this.ids.submitButtonId);
    }
    async click_cancel() {
        await this.page.click(this.ids.cancelButtonId);
    }
    async click_logout() {
        await this.page.click(this.ids.logoutButtonId);
    }
    async add_sample_contact_with_index(version_index) {
        await this.enter_first_name(`firstNamev${version_index}`);
        await this.enter_last_name(`lastNamev${version_index}`);
        await this.enter_email(`emailv${version_index}@q.com`);
        await this.enter_birthdate('2023-10-01');
        await this.enter_phone('1234567890');
        await this.enter_street1(`street1v${version_index}`);
        await this.enter_street2(`street2v${version_index}`);
        await this.enter_city(`cityv${version_index}`);
        await this.enter_state(`statev${version_index}`);
        await this.enter_postal_code('12345');
        await this.enter_country(`countryv${version_index}`);
    }
    async get_error_message() {
        return await this.page.textContent(this.ids.errorMessageId);
    }
}
module.exports = ContactListPage;

File: aqa_python/automation_playwright/pom/addcontact_page.py
from pom.base_page import BasePage

class AddContactPage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.page_url = f"{self.config.BASE_URL}/addContact"
        self.selectors = {
            "logout_button": "#logout",
            "first_name": "#firstName",
            "last_name": "#lastName",
            "email": "#email",
            "birthdate": "#birthdate",
            "phone": "#phone",
            "street1": "#street1",
            "street2": "#street2",
            "city": "#city",
            "state": "#stateProvince",
            "postal_code": "#postalCode",
            "country": "#country",
            "submit_button": "#submit",
            "cancel_button": "#cancel",
            "error_message": "#error"
        }

    def navigate(self):
        self.go_to_url(self.page_url)
        self.page.wait_for_timeout(200)

    def enter_first_name(self, value: str):
        self.page.fill(self.selectors["first_name"], value)
        self.page.wait_for_timeout(200)

    def enter_last_name(self, value: str):
        self.page.fill(self.selectors["last_name"], value)
        self.page.wait_for_timeout(200)

    def enter_email(self, value: str):
        self.page.fill(self.selectors["email"], value)
        self.page.wait_for_timeout(200)

    def enter_birthdate(self, value: str):
        self.page.fill(self.selectors["birthdate"], value)
        self.page.wait_for_timeout(200)

    def enter_phone(self, value: str):
        self.page.fill(self.selectors["phone"], value)
        self.page.wait_for_timeout(200)

    def enter_street1(self, value: str):
        self.page.fill(self.selectors["street1"], value)
        self.page.wait_for_timeout(200)

    def enter_street2(self, value: str):
        self.page.fill(self.selectors["street2"], value)
        self.page.wait_for_timeout(200)

    def enter_city(self, value: str):
        self.page.fill(self.selectors["city"], value)
        self.page.wait_for_timeout(200)

    def enter_state(self, value: str):
        self.page.fill(self.selectors["state"], value)
        self.page.wait_for_timeout(200)

    def enter_postal_code(self, value: str):
        self.page.fill(self.selectors["postal_code"], value)
        self.page.wait_for_timeout(200)

    def enter_country(self, value: str):
        self.page.fill(self.selectors["country"], value)
        self.page.wait_for_timeout(200)

    def click_submit(self):
        self.page.click(self.selectors["submit_button"])
        self.page.wait_for_timeout(200)

    def click_cancel(self):
        self.page.click(self.selectors["cancel_button"])
        self.page.wait_for_timeout(200)

    def click_logout(self):
        self.page.click(self.selectors["logout_button"])
        self.page.wait_for_timeout(200)

    def add_sample_contact_with_index(self, index: int):
        self.enter_first_name(f"firstNamev{index}")
        self.enter_last_name(f"lastNamev{index}")
        self.enter_email(f"emailv{index}@q.com")
        self.enter_birthdate("2023-10-01")
        self.enter_phone("1234567890")
        self.enter_street1(f"street1v{index}")
        self.enter_street2(f"street2v{index}")
        self.enter_city(f"cityv{index}")
        self.enter_state(f"statev{index}")
        self.enter_postal_code("12345")
        self.enter_country(f"countryv{index}")

    def get_error_message(self) -> str:
        return self.page.text_content(self.selectors["error_message"])
